### DISCUSSION TITLE
Window Positioning: How to make Kitty open in the center of the screen

### DISCUSSION THREAD

**Discussion by weiameili** [kovidgoyal/kitty#3412] [Q&A]
Hi, I want kitty to open in the center of the screen, instead of a corner. How would I achieve the desired effect?

The `placement_strategy` config option does not have any effect and when opening kitty from the terminal I get:
`[081 17:56:02.563230] Ignoring unknown config key: placement_strategy`

**Comment by kovidgoyal** [comment]
OS Window positioning is done my window managers, not applications. placement_strategy is about how to position the extra space *inside* a kitty window ehn the window size is not a multiple of the cell size.

**Reply by weiameili** [reply]
Hi, I am using Deepin 20 with the default DE, so I don't really have the option to specify where a new kitty window opens through my WM. Doesn't kitty have an option to control window position like other terminals, say URxvt.

**Comment by kovidgoyal** [comment]
No, it doesnt. Such an option wouldnt work on many window managers such as tiling ones, or any window manager at all in wayland where applications are not allowed to set window position at all.

**Reply by BMJ-pdx** [reply]
But roxterm, mate-terminal, gnome-terminal and xterm can all specify a _preferred_ (or suggested) geometry (via command-line option) that can optionally include a position.  (Granted, the CLI geometry options may be X11-specific.)
I find this vital -- when I log in a script starts about a dozen terms, typically two per (MATE) workspace, non-overlapping, left and right.  Multiple kitty windows don't provide the same functionality (e.g. being able to minimize one of a pair of term/OS windows, to expose more desktop).

**Comment by deccer** [comment]
I know this is an old issue. But it also kind of annoys me when windows open at random locations on non titling window managers, no matter what OS.

Would it be possible to get such an option though. I personally don't care about wayland enough yet either. perhaps an x11/winapi related override, similar to the OS-specific settings?

Anywho. Thank you for kitty :)

**Comment by markcst** [comment]
So no solution on this? As mentioned by @BMJ-pdx, other terminals like gnome-terminal (default terminal in Linux Mint Cinnamon) can do this, so maybe it's technically possible?

**Reply by kovidgoyal** [reply]
On X11 without a tiling window manager ONLY.

**Reply by Neurognostic** [reply]
If your window manager does not provide facilities for window rules. You could wrap launching kitty with a shell script and use a tool like `xdotool` to move it's window to a specified position as soon as it is launched.

**Reply by CartConnoisseur** [reply]
> On X11 without a tiling window manager ONLY.

so, "ONLY" the *vast* majority of desktop linux users?

**Comment by markcst** [comment]
I'm currently using devilspie2 to achieve this

**Comment by mxz7** [comment]
i'm using macos without a special window manager and it opens seemingly in a random place every time, i have to drag it to the center.

**Comment by JulioXoX** [comment]
Looking for solution too.

**Comment by tadg-paul** [comment]
The trouble is that Kitty does not behave on the MacOS windowing system in the way that other apps do, and I'm not entirely sure why.

I use BetterTouchTool to manage my windowing on MacOS, and Kitty is the only application that does not respond to instructions to move or to resize. I also tried scripting it with Hammerspoon - Kitty behaves in unexpected ways (it will sometimes but not always move the window, and never resizes it).

I had been using Kitty for a number of years - and managing its windows with BTT and Hammerspoon without any problems.
One day, I don't recall exactly which release, it stopped behaving.
At the time I was under a project deadline, so did not have much time to spend on figuring it out or logging an issue. I switched to using iTerm and Alacritty for the sake of expediency.

Every now and then though I give Kitty another try - I do miss this really fantastic terminal emulator, but its inability to play nice on the MacOS windowing system is unfortunately a deal breaker for me.
